python/cucim/src/cucim/core/operations/intensity/zoom.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import math
+from typing import Any, List, Optional, Sequence, Tuple, Union
+
+import cupy
+import numpy as np
+
+_logger = logging.getLogger("zoom_cucim")
+
+from .kernel.cuda_kernel_source import cuda_kernel_code
+
+CUDA_KERNELS = cupy.RawModule(code=cuda_kernel_code)
+
+def zoom(
+    img: Any,
+    zoom_factor: Sequence[float]
+    ):
+    """Zooms an ND image
+
+    Parameters
+    ----------
+    img : channel first, cupy.ndarray or numpy.ndarray
+        Input data. Can be numpy.ndarray or cupy.ndarray
+    zoom_factor: Sequence[float]
+        The zoom factor along the spatial axes.
+        Zoom factor should contain one value for each spatial axis.
+    Returns
+    -------
+    out : cupy.ndarray or numpy.ndarray
+        Output data. Same dimensions and type as input.
+
+    Raises
+    ------
+    TypeError
+        If input 'img' is not cupy.ndarray or numpy.ndarray
+
+    Examples
+    --------
+    >>> import cucim.core.operations.intensity as its
+    >>> # input is channel first 3d array
+    >>> output_array = its.zoom(input_arr,[1.1,1.1])
+    """
+    try:
+        if len(img.shape) == 4:
+            N, C, H, W = img.shape
+        elif len(img.shape) == 3:
+            C, H, W = img.shape
+            N = 1
+
+        output_size_cu = [N, C, int(math.floor(H*zoom_factor[0])), int(math.floor(W*zoom_factor[1]))]
+
+        if output_size_cu[2] == H and output_size_cu[3] == W:
+            return img
+
+        to_cupy = False
+        mempool = cupy.get_default_memory_pool()
+
+        cupy_img = img
+        if isinstance(img, np.ndarray):
+            to_cupy = True
+            cupy_img = cupy.asarray(img.astype(img.dtype))
```

Comment:
```suggestion
        if isinstance(img, np.ndarray):
            cupy_img = cupy.asarray(img, order="C")
        else:
            cupy_img = cupy.ascontiguousarray(img)
```
ensure kernel input is C-contiguous